addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -196,6 +198,40 @@ def _update_invoice_from_xml_tree(self, filename, tree, invoice):
                 return self._import_fattura_pa(tree, invoice)
         return super()._update_invoice_from_xml_tree(filename, tree, invoice)
 
+    def _decode_p7m_to_xml(self, filename, content):
+        decoded_content = remove_signature(content)
+        if not decoded_content:
+            return False
+
+        try:
+            # Some malformed XML are accepted by FatturaPA, this expends compatibility
+            parser = etree.XMLParser(recover=True)
+            xml_tree = etree.fromstring(decoded_content, parser)
+        except Exception as e:
+            _logger.exception("Error when converting the xml content to etree: %s" % e)
+            return False
+        if not len(xml_tree):
+            return False
+
+        return xml_tree
+
+    def _create_invoice_from_binary(self, filename, content, extension):
+        self.ensure_one()
+        if extension.lower() == '.xml.p7m':
+            decoded_content = self._decode_p7m_to_xml(filename, content)
+            if len(decoded_content) and self._is_fattura_pa(filename, decoded_content):
```

Comment:
`_decode_p7m_to_xml` can retun `False` so don't use `len` here (same below)

Reply:
And since `_decode_p7m_to_xml` is an internal method (so doesn't have to be RPC-compatible) it should probably return `None` rather than `False`.